File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/adapter/RoleRepositoryAdapter.java
package spring.web.java.infrastructure.persistence.adapter;

import java.util.Optional;

import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.model.entity.Role;
import spring.web.java.domain.repository.RoleRepository;
import spring.web.java.infrastructure.persistence.RoleJpaRepository;

@Component
@RequiredArgsConstructor
public class RoleRepositoryAdapter implements RoleRepository {

	private final RoleJpaRepository jpaRepository;

	@Override
	public Role save(Role role) {
		return jpaRepository.save(role);
	}

	@Override
	public Optional<Role> findById(Long id) {
		return jpaRepository.findById(id);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/RoleRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Component
import spring.web.kotlin.domain.model.entity.Role
import spring.web.kotlin.domain.repository.RoleRepository
import spring.web.kotlin.infrastructure.persistence.RoleJpaRepository

@Component
class RoleRepositoryAdapter(
    private val jpaRepository: RoleJpaRepository
) : RoleRepository {
    override fun save(role: Role) = jpaRepository.save(role)

    override fun findByIdOrNull(id: Long) = jpaRepository.findByIdOrNull(id)
}